RecoLocalTracker/SiStripClusterizer/plugins/alpaka/SiStripClusterizerConditionsESProducerAlpaka.cc

```diff
@@ -0,0 +1,225 @@
+  public:
+    SiStripClusterizerConditionsESProducerAlpaka(edm::ParameterSet const& iConfig) : ESProducer(iConfig) {
+      // Two tokens for quality because of the exc message:
+      // You may need multiple tokens if you want to get the same data in multiple transitions.
+      auto listenA = setWhatProduced(this, &SiStripClusterizerConditionsESProducerAlpaka::produceDetToFeds);
+      qualityTokenA_ = listenA.consumes(iConfig.getParameter<edm::ESInputTag>("QualityLabel"));
+
+      auto listenB = setWhatProduced(this, &SiStripClusterizerConditionsESProducerAlpaka::produceData);
+      qualityTokenB_ = listenB.consumes(iConfig.getParameter<edm::ESInputTag>("QualityLabel"));
+      gainsToken_ = listenB.consumes();
+      noisesToken_ = listenB.consumes();
+    }
+
+    static void fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+      edm::ParameterSetDescription desc;
+      desc.add<edm::ESInputTag>("QualityLabel");
+      desc.add<edm::ESInputTag>("Label");
+      descriptions.addWithDefaultLabel(desc);
+    }
+
+    std::unique_ptr<SiStripClusterizerConditionsDetToFedsHostObject> produceDetToFeds(
+        SiStripClusterizerConditionsDetToFedsRecord const& iRecord) {
+      const auto& quality = iRecord.get(qualityTokenA_);
+
+      auto product = std::make_unique<SiStripClusterizerConditionsDetToFedsHostObject>(cms::alpakatools::host());
+      product->zeroInitialise();
+
+      auto& detToFeds_qualityFlags = (*product).data()->qualityOk;
+
+      // connected: map<DetID, std::vector<int>>
+      // map of KEY=detid DATA=vector of apvs, maximum 6 APVs per detector module :
+      const auto& connected = quality.cabling()->connected();
+      // detCabling: map<DetID, std::vector<const FedChannelConnection*>
+      // map of KEY=detid DATA=vector<FedChannelConnection>
+      const auto& detCabling = quality.cabling()->getDetCabling();
+
+      for (const auto& conn : connected) {
+        const auto det = conn.first;
+        if (!quality.IsModuleBad(det)) {
+          const auto detConn_it = detCabling.find(det);
+
+          if (detCabling.end() != detConn_it) {
+            for (const auto& conn : (*detConn_it).second) {
+              if (conn && conn->fedId() && conn->isConnected()) {
+                const auto conn_fedID = conn->fedId();
+                const auto conn_fedCh = conn->fedCh();
+
+                uint32_t idx = channelIndex(conn_fedID, conn_fedCh);
+                detToFeds_qualityFlags[idx] = true;
+                // ss << idx << ",";
+              }
+            }
+          }
+        }
+      }
+
+      return product;
+    }
+
+    std::unique_ptr<SiStripClusterizerConditionsDataHostObject> produceData(
+        SiStripClusterizerConditionsDataRecord const& iRecord) {
+      auto gains = iRecord.getTransientHandle(gainsToken_);
+      const auto& noises = iRecord.get(noisesToken_);
+      const auto& quality = iRecord.get(qualityTokenB_);
```

Comment:
Curiosity: why `gains` is a transient handle, while `noises` and `quality` are handles ?